launch/launch/launch_frontend/entity.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for Entity class."""
+
+from typing import List
+from typing import Optional
+from typing import Text
+from typing import Type
+from typing import Union
+
+from .type_utils import SomeAllowedTypes
+
+
+class Entity:
+    """Single item in the intermediate front_end representation."""
+
+    @property
+    def type_name(self) -> Text:
+        """Get Entity type."""
+        raise NotImplementedError()
+
+    @property
+    def parent(self) -> Optional['Entity']:
+        """Get Entity parent."""
+        raise NotImplementedError()
+
+    @property
+    def children(self) -> List['Entity']:
+        """Get the Entity's children."""
+        raise NotImplementedError()
+
+    def get_attr(
+        self,
+        name: Text,
+        *,
+        types:
+            Optional[
+                Union[
+                    SomeAllowedTypes,
+                    Type[List['Entity']],
+                ]
+            ] = str,
+        optional: bool = False
+    ) -> Optional[Union[
+        Text,
+        int,
+        float,
+        List[Text],
+        List[int],
+        List[float],
+        List['Entity']
+    ]]:
+        """
+        Access an attribute of the entity.
+
+        By default, it will try to return it as an string.
+        `types` states the expected types of the attribute. Type coercion or type checking is
+        applied depending on the particular frontend.
+
+        The allowed types are:
+            - `str`
+            - `int`
+            - `float`
+            - `bool`
+            - `List[str]`
+            - `List[int]`
+            - `List[float]`
+            - `List[bool]`
+
+        Types that can not be combined with the others:
+            - `List[Entity]`
+
+        `types = None` work in the same way as:
+            `(int, float, bool, List[int], List[float], List[bool], List[str], str)`
+        `List[Entity]` will return a list of more entities.
+
+        See the frontend documentation to see how `list` and `List[Entity]` look like for each
+        frontend implementation.
+
+        If `optional` argument is `True`, will return `None` instead of raising `AttributeError`.
```

Comment:
@ivanpauno consider
```suggestion
        If `optional` is `True` and the attribute cannot be found, `None` will be returned instead of raising `AttributeError`.
```